tests/batch/indexer_Position_Bond_test.py

```diff
@@ -1048,32 +1066,222 @@ def test_normal_17(self, processor, shared_contract, session):
     ###########################################################################
     # Error Case
     ###########################################################################
-
-    # <Error_1>
-    # Error occur
-    @mock.patch("web3.contract.ContractEvent.getLogs", MagicMock(side_effect=Exception()))
-    def test_error_1(self, processor, shared_contract, session):
+    # <Error_1_1>: ABIEventFunctionNotFound occurs in __sync_xx method.
+    # <Error_1_2>: ServiceUnavailable occurs in __sync_xx method.
+    # <Error_2_1>: ServiceUnavailable occurs in "initial_sync" / "sync_new_logs".
+    # <Error_2_2>: SQLAlchemyError occurs in "initial_sync" / "sync_new_logs".
+    # <Error_3>: ServiceUnavailable occurs and is handled in mainloop.
+
+    # <Error_1_1>: ABIEventFunctionNotFound occurs in __sync_xx method.
+    @mock.patch("web3.contract.ContractEvent.getLogs", MagicMock(side_effect=ABIEventFunctionNotFound()))
+    def test_error_1_1(self, processor, shared_contract, session):
         # Issue Token
         token_list_contract = shared_contract["TokenList"]
         personal_info_contract = shared_contract["PersonalInfo"]
         token = self.issue_token_bond(
             self.issuer, config.ZERO_ADDRESS, personal_info_contract["address"], token_list_contract)
         self.listing_token(token["address"], session)
-
         PersonalInfoUtils.register(
             self.trader["account_address"], personal_info_contract["address"], self.issuer["account_address"])
+        # Transfer
+        bond_transfer_to_exchange(
+            self.issuer, {"address": self.trader["account_address"]}, token, 10000)
+
+        block_number_current = web3.eth.blockNumber
+        # Run initial sync
+        processor.initial_sync()
+
+        # Assertion
+        _position_list = session.query(IDXPosition).order_by(IDXPosition.created).all()
+        assert len(_position_list) == 0
+        # Latest_block is incremented in "initial_sync" process.
+        _idx_position_bond_block_number = session.query(IDXPositionBondBlockNumber).first()
+        assert _idx_position_bond_block_number.latest_block_number == block_number_current
 
         # Transfer
         bond_transfer_to_exchange(
             self.issuer, {"address": self.trader["account_address"]}, token, 10000)
 
+        block_number_current = web3.eth.blockNumber
         # Run target process
         processor.sync_new_logs()
 
         # Run target process
         processor.sync_new_logs()
 
+        # Clear cache in DB session.
+        session.rollback()
```

Comment:
If using postgresql, expire_all() is sufficient to clear the cache in the session instance.
But with MySQL, rollback() must be called to do so.
I think that’s because they have different transaction isolation level.

https://docs.sqlalchemy.org/en/14/orm/session_api.html#:~:text=the%20given%20instance.-,Note%20that%20a%20highly%20isolated%20transaction%20will%20return%20the%20same%20values%20as%20were%20previously%20read%20in%20that%20same%20transaction%2C%20regardless%20of%20changes%20in%20database%20state%20outside%20of%20that%20transaction.,-To%20expire%20individual

Reply:
https://docs.sqlalchemy.org/en/14/faq/sessions.html#faq-session-identity
```
But remember, the ORM cannot see changes in rows if our isolation level is repeatable read or higher, unless we start a new transaction.
```